File: game-server/src/main/java/org/dm/event/widget/DialogueEvent.java
package org.dm.event.widget;

import com.dm.game.world.entity.mob.player.Player;

/**
 * @author Jire
 */
public final class DialogueEvent implements WidgetEvent {

    public static final DialogueEvent INSTANCE = new DialogueEvent();

    private DialogueEvent() {
        // Private constructor for singleton
    }

    @Override
    public void handle(Player player) {
        player.dialogueFactory.execute();
    }

}

File: game-server/src/main/kotlin/org/jire/tarnishps/event/widget/DialogueEvent.kt
package org.jire.tarnishps.event.widget

import com.dm.game.world.entity.mob.player.Player

/**
 * @author Jire
 */
object DialogueEvent : WidgetEvent {

    override fun handle(player: Player) {
        player.dialogueFactory.execute()
    }

}